### PR TITLE
WIP: Add script to trivialize some left-wilcard targets

### PR DESCRIPTION
**Pull request by cschanaj**
This script will trivialize at least 1.3k left-wildcard targets (nearly 1/3 of their population). In some sense, it seems too good to be true, hope I didn't mess up anything :smile: 

P.S. benchmark with memusage.js show the memory usage will be increased by 0.6 MB because of this change... if this is not justified, I will close this PR.

### CODE REVIEW

**Comment by pipboy96** [comment]
@cschanaj This is justified since it reduces ruleset complexity and improves ruleset testing coverage, which are currently very important tasks.

File: utils/trivialize-rules/trivialize-targets.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    // note that "trivial" does not mean a trivial rule here
+    function isTrivialRewrite(rule) {
+      let retval = true;
+
+      try {
+        explodeRegExp(rule.from, url => {
+          if (validUrl.isUri(url)) {
+            let domain = new URL(url).hostname;
+            rewriteDomains.add(domain);
+            retval &= true;
```
Comment (pipboy96) [inline utils/trivialize-rules/trivialize-targets.js]: This is a no-op since any of `[true, false]` when ran through `a => a && true` will stay identical.

**Comment by pipboy96** [comment]
Also, the actual `package.json` is missing, even though `package-lock.json` is there.

File: utils/trivialize-rules/trivialize-targets.js [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+          } else {
+            retval = false;
+          }
+        });
+      } catch (e) {
+        // assume it is a non-trivial rewrite here
```
Comment (pipboy96) [inline utils/trivialize-rules/trivialize-targets.js]: Can you add a logger here and see if this gets executed, and what triggers it?

**Comment by pipboy96** [comment]
Also makes sense to replace rules that are effectively equal to `^http:` :arrow_right: `https:` after these changes. Will try to implement that.

**Comment by cschanaj** [comment]
> Also makes sense to replace rules that are effectively equal to `^http:`  `https:` after these changes. Will try to implement that.

AFAIK, `trivialize-rules.js` already do part of this. It might be helpful if you would like to implement that yourself.

**Pull request closed without merging**